exchanges/orderbook/depth.go

```diff
@@ -262,17 +313,23 @@ func (d *Depth) GetName() string {
 }
 
 // IsRestSnapshot returns if the depth item was updated via REST
-func (d *Depth) IsRestSnapshot() bool {
+func (d *Depth) IsRestSnapshot() (bool, error) {
 	d.m.Lock()
 	defer d.m.Unlock()
-	return d.restSnapshot
+	if !d.isValid {
+		return false, ErrOrderbookInvalid
```

Comment:
Yeah, it might be misleading when we don't change the state, but even if somebody ignores the error there shouldn't be data in memory anymore. I could turn the bool into a pointer and flush value but I didn't see a point(er). It should be invalid until either a websocket snapshot comes through or a REST snapshot comes through.